src/v/s3/signature.cc

```diff
@@ -0,0 +1,392 @@
+#include "s3/signature.h"
+
+#include "s3/error.h"
+#include "s3/logger.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <boost/algorithm/string/compare.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <gnutls/crypto.h>
+#include <gnutls/gnutls.h>
+
+#include <memory>
+#include <system_error>
+
+namespace s3 {
+
+constexpr auto algorithm = "AWS4-HMAC-SHA256";
```

Comment:
nit: we don't typically use constexpr auto. i'd suggest std::string_view

Reply:
also i think in c++20 std::string and std::string_view can be constexpr :)